packages/common/src/storage/helpers/storage-deserializer-maker.ts

```diff
@@ -0,0 +1,57 @@
+import BigNumber from 'bignumber.js'
+
+export enum StorageReviverType {
+  AsBigInt = 'AsBigInt',
+  AsBigNumber = 'AsBigNumber',
+}
+
+export type StorageReviverMapping = {
+  [key: string]: StorageReviverType
+}
+
+/**
+ * Creates a storage deserializer function based on the provided mapping.
+ * Since BigInt and BigNumber are not supported by JSON.parse, it stores as strings and revive as BigInt or BigNumber.
+ *
+ * @param mapping - The mapping of keys to reviver types.
+ * @returns JSON object revived according to the mapping, string -> BigInt or BigNumber.
+ */
+export const storageDeserializerMaker = (mapping: StorageReviverMapping) => {
```

Comment:
serializing is simple, by finding BigInt/BigNumber properties, for deserializing a map is needed, it doesn't take paths, so if `x` is a mapped, any `x` level will be considered whether BI/BN, therefore bigint names **must** be unique. eg. `a.b.c.x` x = BN/BI, also for arrays it is needed to embed in a object.